src/custom/state/claim/hooks/index.ts

```diff
@@ -86,14 +86,34 @@ export type RepoClaims = RepoClaimData[]
 /**
  * Gets an array of available claim
  *
+ * By default, it'll filter out claims that are no longer available when the time window is closed
+ * It's possible to ignore the time window and return all unclaimed claims
+ *
  * @param account
+ * @param ignoreTimeWindow
  */
-export function useUserAvailableClaims(account: Account): UserClaims {
+export function useUserAvailableClaims(account: Account, ignoreTimeWindow = false): UserClaims {
   const userClaims = useUserClaims(account)
   const contract = useVCowContract()
 
+  const isInvestmentStillAvailable = useInvestmentStillAvailable()
+  const isAirdropStillAvailable = useAirdropStillAvailable()
+
   // build list of parameters, with the claim index
-  const claimIndexes = useMemo(() => userClaims?.map(({ index }) => [index]) || [], [userClaims])
+  const claimIndexes = useMemo(() => {
```

Comment:
The combination of indexes and indices is melting my brain stem 🤯

Reply:
![](https://external-content.duckduckgo.com/iu/?u=http%3A%2F%2F2.bp.blogspot.com%2F-HPOnodccnXg%2FVZZ-ZvCW9PI%2FAAAAAAAAAwM%2FzFod5L1uepU%2Fs1600%2FBrain%252BMush.jpg&f=1&nofb=1)